src/riak_kv_vnode.erl

```diff
@@ -2041,6 +2041,60 @@ result_fun_ack(Bucket, Sender) ->
             end
     end.
 
+%% ------------------------------------------------------------
+%% Specialization of result_fun_ack for range scans.  
+%%
+%% This version checks the cluster capabilities for {riak_kv,
+%% decode_query_results_at_vnode}.
+%%
+%% If true (i.e., we have an upgraded cluster where all nodes support
+%% parallel decoding), we reply with the query results already
+%% decoded, so that the work of decoding the results can be
+%% parallelized among the vnodes in the query coverage plan, instead
+%% of being serialized at the coordinating vnode.
+%%
+%% If false (i.e., we have a mixed-mode cluster), we reply with the
+%% encoded results, as before.
+%% ------------------------------------------------------------
+
+range_scan_result_fun_ack(Bucket, Sender) ->
+    fun(Items) ->
```

Comment:
A small feature request, or I can do it after if you prefer. Can this fun call a function which does the heavy lifting, instead of inside the fun makes it traceable, and so debuggable. I had to do this for my `lists:flatten/1` investigation above.
```erlang
range_scan_result_fun_ack_fun(Bucket, Sender) ->
    fun(Items) ->
        range_scan_result_fun_ack(Bucket, Sender, Items)
    end.

range_scan_result_fun_ack(Bucket, Sender, Items) ->
```